### DISCUSSION TITLE
bin file für OpenDTU ?

### DISCUSSION THREAD

**Discussion by mroenne2022** [tbnobody/OpenDTU#208] [Q&A]
Hallo, 
ich bin noch ziemlich neu hier. Ich würde gerne versuchen mir einen OpenDTU selbst herzustellen um meinen
HM-600 WR auszulesen, bzw. die erzeugte Leistung zu sehen.
Entsprechende HW hab ich schon hier ( ESP-WROOM-32 / 30Pins ) und einen NRF24L01+ Board.
Das Flashtool NodeMCU-PyFlasher hab ich schon , nun bin ich auf der Suche nach einem fertigen
bin File für eine OpenDTU ? 
Vielleicht kann jemand helfen und könnte eines bereitstellen ?
VG Michael R.

**Comment by tbnobody** [comment]
Hallo,
verwende doch einfach die die hier erzeugt werden: https://github.com/tbnobody/OpenDTU/actions/workflows/build.yml
Auch oben im Menü unter "Actions" erreichbar

**Reply by mroenne2022** [reply]
OK , Danke Dir. 
War mir so nicht klar wie das funktioniert - bin wie gesagt neu in diesem Umfeld. 
Hab jetzt das generic zip file gefunden mit den bin Dateien.

**Reply by mpl1337** [reply]
wo finde ich das zip? suche mir schon nen wolf.... VSCODE funktioniert auch nicht mehr bei mir :/

**Reply by mroenne2022** [reply]
Hallo mpl1337 ,
ging mir zuerst genauso.
Einfach diesem Link folgen : https://github.com/tbnobody/OpenDTU/actions/workflows/build.yml
Dann Links auf OpenDTU Build , dann auf einen dieser webapp: added ..... klicken
Neue Seite öffnet sich - dann ganz unten liegt ein Zip File zum download bereit : opendtu-generic 
Dort sind die 4 bin Files enthalten.
Ich habs aber damit nicht weiter verfolgt , hab jetzt meinen ESP32 mit Ahoy DTU Firmware geflasht, hab aber da noch Probleme mit dem NRF Modul bzgl. PIN/Verkabelung.
VG

**Reply by Franky806** [reply]
Hallo,
welche von den 4 bin`s muß ich flashen?

**Reply by madmartin** [reply]
> Hallo, welche von den 4 bin`s muß ich flashen?

Beim ersten Flash mußt Du alle 4 flashen, an verschiedene Adressen. Da ist eine Tabelle im README.md wo diese Adressen stehen. 
Wenn OpenDTU dann mal läuft, gibt es in der Web-Oberfläche einen Menüpunkt "Firmware Upgrade", da muß dann die Datei "opendtu-generic.bin" ausgewählt werden.
Ich such gleich mal ein Beispiel raus fürs flashen mit esptool.py

**Reply by madmartin** [reply]
Beispiel mit esptool.py:

```
esptool.py --port /dev/ttyUSB0 --chip esp32 --before default_reset --after hard_reset \
  write_flash --flash_mode dout --flash_freq 40m --flash_size detect \
  0x1000 bootloader_dio_40m.bin \
  0x8000 partitions.bin \
  0xe000 boot_app0.bin \
  0x10000 opendtu-generic.bin
```


@tbnobody das sollten wir mal ins README.md aufnehmen.

**Reply by Franky806** [reply]
Super Danke dir schonmal

**Reply by Franky806** [reply]
Hab es mit ESP32 Flash Download tool probiert, läuft auch Einwandfrei durch aber ich finde kein dtu Netzwerk ?

**Reply by madmartin** [reply]
>  ich finde kein dtu Netzwerk ?

Meinst Du es erscheint der AccessPoint "OpenDTU-*" nicht?
Meine Tips:
Vor dem ersten Flash 1x komplettes "erase flash" machen.
Ansonsten grundsätzlich bei solchen Problem: Mit einem Terminal-Programm (z.B. "putty") auf das serielle Device drauf (Geschwindigkeit 115k, 8n1) und das serielle Log anschauen. Mal die Reset-Taste an der ESP32-Platine drücken, damit man alles von anfang an sehen kann.
Ggf. das seriell Log hier mal posten.

Ich habe ferner festgestellt daß der Flash Mode und die Frequenz eine Rolle spielen. Stimmt das nicht, ist zwar der Flash-Vorgang an sich erfolgreich, aber das Programm läuft nicht. Erkennt man dann am serial log.
Schau mal ob Du beim "ESP32 Flash Download tool" solche Einstellungen findest. Funktioneren tut "flash_mode dout" und "flash_freq 40m" (siehe oben esptool.py)

**Reply by Franky806** [reply]
Super Vielen Dank es läuft. Deine Tips haben geholfen

**Reply by HolgiB** [reply]
Danke für die detailierten Erklärungen. Ich hab mich zwar auch ein bisschen "gequält" mit dem Suchen nach der fertigen Bin-Files aber das Flashen ging und so besser. :-)

**Reply by tbnobody** [reply]
Aber die liegen doch jetzt sogar direkt auf der Startseite und ohne github account? 😥

![image](https://user-images.githubusercontent.com/1753222/233466869-a475c3d2-add2-4595-a289-04a3f6532871.png)

**Reply by HolgiB** [reply]
Hm, vielleicht war ich einfach zu doof. Ich bin direkt auf die Haupt Git-Seite des Projektes:
https://github.com/tbnobody/OpenDTU

Und dort hab ich in der Übersicht "nur" die Anleitung und den Quellcode gefunden. Vermutlich bin ich einfach nur noch so Git-affin. Mach dir keinen Kopf :-)

**Reply by madmartin** [reply]
Die bin-files liegen unter "Releases":

![grafik](https://user-images.githubusercontent.com/4208120/233488449-a5a2f047-0466-4096-9416-3ce36fd8775f.png)

**Comment by ChaosChris1975** [comment]
Hallo und guten Abend.
Erstmal muss ich ein großes Lob an dieses Projekt hier loswerden.
Aber man muss sich mit GitHub wohl erst anfreunden. :-)
Habe auch etwas gebraucht, um die .bin Files zu finden. Und mit dem Flashtool von Espressif geht es super easy.
Habe eben über die Oberfläche die aktuelle opendtu-generic.bin hochgeladen. Super Sache. Aber eine Sicherheitsabfrage nach dem auswählen der Datei wäre vielleicht sinnvoll. Hoffe da wird auf plausibilität geprüft und nicht einfach alles hochgeladen, was man eventuell falsch ausgewählt hat.

**Reply by ff4-solar** [reply]
Irgendwie habe ich kein Glück und find die bin-Files nicht.
Liegt wohl daran dass ich mich noch nicht so gut auskenne.

Es wäre toll wenn mir jemand einen Wink mit dem Zaunpfahl geben könnte, wo die bin-Files liegen
Ich finde hier leide nichts :-(
https://github.com/tbnobody/OpenDTU/actions/workflows/build.yml

**Reply by ChaosChris1975** [reply]
Deinem Link folgen, oberste "webapp....." anklicken, dann ganz unten auf der Seiten die Datei "opendtu-generic" downloaden, dort sind die bin-Files gepackt.
Bin mir nicht sicher, aber es könnte sein das du dazu eingeloggt sein musst.

**Reply by ff4-solar** [reply]
Danke Chris,
für den schnelle Service !!!
Das muss man wirklich erstmal  wissen :-)

**Reply by ChaosChris1975** [reply]
Bitte, hab auch lange gesucht....

**Comment by FrodoVDR** [comment]
Eine Flash Software für Windows und Macs gibt es hier [https://github.com/Jason2866/ESP_Flasher/releases/tag/v.1.4.1](url)

**Reply by Mike223311** [reply]
sorry, page not found!

**Reply by Solarteur** [reply]
Here is an alternative:
https://github.com/tbnobody/OpenDTU#flash-with-espressif-flash-download-tool-windows

**Reply by Mike223311** [reply]
merc'i
